cmd/restic/cmd_cat.go

```diff
@@ -33,9 +33,35 @@ func init() {
 	cmdRoot.AddCommand(cmdCat)
 }
 
+func getPipeSeperatedTypes(types []string) (str string) {
+	for i, t := range types {
+		if i != 0 {
+			str += "|"
+		}
+		str += t
+	}
+	return
+}
+
 func runCat(ctx context.Context, gopts GlobalOptions, args []string) error {
-	if len(args) < 1 || (args[0] != "masterkey" && args[0] != "config" && len(args) != 2) {
-		return errors.Fatal("type or ID not specified")
+	allowedTypes := []string{"config", "index", "snapshot", "key", "masterkey", "lock", "pack", "blob", "tree"}
```

Comment:
please extract the parameter validation into a separate function that's called by `runCat`.